src/libopensc/card-openpgp.c

```diff
@@ -2948,15 +2933,13 @@ pgp_store_key(sc_card_t *card, sc_cardctl_openpgp_keystore_info_t *key_info)
 	r = pgp_update_pubkey_blob(card, key_info->rsa.n, 8*key_info->rsa.n_len,
 	                           key_info->rsa.e, 8*key_info->rsa.e_len, key_info->key_id);
 
-	sc_log(ctx, "Update card algorithms.");
+	sc_log(card->ctx, "Update card algorithms.");
 	pgp_update_card_algorithms(card, &pubkey);
 
-out:
-	if (data) {
+err:
+	if (data != NULL)
```

Comment:
this check is technically not needed, `free()` works on `NULL`

Reply:
@frankmorgner  issue addressed & fixed with the latest commit